crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -276,17 +285,16 @@ impl CheckpointExecutor {
         let num_tasks_to_schedule = std::cmp::min(checkpoints_diff, tasks_diff as u64);
 
         // get all checkpoints to be scheduled, less the last (which we got already above)
-        let range: Vec<u64> = (next_to_exec..(next_to_exec + num_tasks_to_schedule)).collect();
+        let range: Vec<u64> = (next_to_exec..(next_to_exec + num_tasks_to_schedule - 1)).collect();
```

Comment:
So the reason for my not doing that is because I am doing a multi-get from checkpoint store rather than serially reading from disk for potentially many checkpoints, and I wanted to avoid fetching more in that multiget than I needed. That said, I agree that the optimization may not be worth the complexity. I'll make this simpler

Reply:
Actually circling back, I still feel this is worth doing. If we are way behind, then we would have to multiget potentially hundreds/thousands of checkpoints eagerly. Alternatively, we could just eagerly get `max_concurrency` checkpoints more, but that may exceed the actual synced checkpoints. Alternatively, we could just serially fetch from disk, but that will definitely slow down things if we are far behind.. 

Currently this is all just some arithmetic. We only need to get it right once. I think I got rid of the real source of complexity that was not worth the performance improvement - namely the append that this bug fixes

Reply:
My experience is that scanning is usually faster than multi get. But the way typed store implement iterators can definitely have unexpected issues. Also for now I think we should lean towards simplicity.

Also, why is the `- 1` needed here?

Reply:
Fair enough, will simplify this and just do a sequential scan and break. And the - 1 was to fix the arithmetic since we are no longer appending the checkpoint from state sync to the end. This will go away with the sequential scan

Reply:
Sounds good, sequential get should be fine for now too. The typed store iter is a bit too inefficient when there are many deletions in the table.